src/Patient_Dashboard/PatientDashboard.js

```diff
@@ -1,60 +1,203 @@
-import React, { useEffect } from 'react';
-import { useParams, useLocation } from 'react-router-dom';
-import DashboardLayout from '../Patient_Dashboard/DashboardLayout';
-import { Search } from 'carbon-components-react';
-import '../Patient_Dashboard/PatientDashboard.scss';
+import React, { useState, useEffect } from "react";
+import { useParams, useLocation, useNavigate } from "react-router-dom";
+import DashboardLayout from "../Patient_Dashboard/DashboardLayout";
+import { Search } from "carbon-components-react";
+import { Document24, Camera24 } from "@carbon/icons-react";
+import "../Patient_Dashboard/PatientDashboard.scss";
 
 const PatientDashboard = () => {
   const { id } = useParams();
+  const navigate = useNavigate();
   const location = useLocation();
   const patientData = location.state ? location.state.patientData : null;
+  const [visits, setVisits] = useState([]);
+  const [selectedDocumentPreviews, setSelectedDocumentPreviews] = useState({});
+  const visitDate = location.state ? location.state.visitDate : null;
+  const [selectedDocument, setSelectedDocument] = useState(null);
+
+  useEffect(() => {
+    fetchVisits();
+  }, []);
+  const fetchVisits = async () => {
+    try {
+      const visitResponse = await fetch(
+        `/openmrs/ws/rest/v1/visit?patient=${id}`
+      );
+      const visitData = await visitResponse.json();
+      setVisits(visitData.results);
+    } catch (error) {
+      console.error("Error fetching visits:", error);
+    }
+  };
+  const formatDate = (inputDate) => {
+    const date = new Date(inputDate);
+    const day = date.getDate();
+    const month = date.toLocaleString("default", { month: "short" });
+    const year = date.getFullYear();
+    return `${day}-${month}-${year}`;
+  };
+  const handleCapture = async () => {
+    if (selectedDocument) {
+      const formData = new FormData();
+      formData.append('file', selectedDocument);
+      try {
+        const uploadResponse = await fetch(
+          '/openmrs/ws/rest/v1/bahmnicore/visitDocument/uploadDocument',
+          {
+            method: 'POST',
+            body: formData,
+          }
+        );
+        if (uploadResponse.ok) {
+          console.log('Document uploaded successfully');
+          setSelectedDocument(null); // Clear the selected document
+          fetchVisits(); // Refresh the visits after upload
+        } else {
+          console.error('Error uploading document');
```

Comment:
There doesn't seem to be code which shows an error to the user in a pop-up / browser. Only console.log is there.